src/App.jsx

```diff
@@ -1,35 +1,106 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import React, { useState } from 'react';
 
-function App() {
-  const [count, setCount] = useState(0)
+function App(){
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [errors, setErrors] = useState({});
+  const [showForm, setShowForm] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const errors = {};
+
+    //first name validation
+    if (!firstName) {
+      errors.firstName = 'First name is required';
+    } else if (firstName.length < 2) {
+      errors.firstName = 'First name must be at least 2 characters long';
+    }
+    //last name validation
+    if (!lastName) {
+      errors.lastName = 'Last name is required';
+    } else if (lastName.length < 2) {
+      errors.lastName = 'Last name must be at least 2 characters long';
+    }
+
+
+    //email validation
+    if (!email) {
+      errors.email = 'Email is required';
+    } else if (!/\S+@\S+\.\S+/.test(email)) {
+      errors.email = 'Invalid email format';
+    }
+
+    if (Object.keys(errors).length === 0) {
+      // Submit the form if there are no errors
+      console.log('Form submitted');
+      setFirstName('');
+      setLastName('');
+      setEmail('');
+    } else {
+      // Set the errors state if there are errors
+      setErrors(errors);
+    }
+  };
+
+  //used to clear form
+  const handleClear = () => {
+    setFirstName('');
+    setLastName('');
+    setEmail('');
+    setErrors({});
+  };
 
   return (
-    <div className="App">
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://reactjs.org" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.jsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
+    <div>
+      <button onClick={() => setShowForm(!showForm)}>
+        {showForm ? 'Hide Form' : 'Show Form'}
+      </button>
+      {showForm && (
```

Comment:
Try to use a ternary here. This works fine in regular React but when we get into React Native, this syntax can cause crashes.

Example:
`{showForm ? ( // JSX code ) : null}`